docs_src/microservices/security/edgex-secrets-config-proxy.md

```diff
@@ -0,0 +1,142 @@
+
+These commands require access to the secret store master key in order to generate temporary secret store access credentials.
+
+# OPTIONS
+
+  * **-m** _/path/to/resp-init.json_ (optional)
+
+    Points to location of secret store master key. Defaults to /vault/config/assets/resp-init.json.
+
+# COMMANDS
+
+  * **tls**
+
+    Configure inbound TLS certificate. This command will configure the TLS secrets into the secret store and re-deploy them to Kong. Requires additional arguments:
+
+    * --certificate _/path/to/certchain.pem_ (required)
+
+      Path to TLS leaf certificate (with optional appended certificate chain, not including a root certificate).
+
+    * --key _/path/to/private.pem_ (required)
+
+      Path to TLS private key.
+
+  * **adduser**
+
+    Create a proxy user using JWT public/private key authentication. Requires additional arguments:
+
+    * --oauth (unimplemented; reserved for future use)
+
+      Create user using OAuth authentication plugin.
+
+    * --jwt (required)
+
+      Create user using JWT authentication plugin.
+
+    * --algorithm HS256 | ES256 (required for JWT method)
+
+      Algorithm used for signing the JWT.
+
+    * --public\_key _public\_key.pem_ (required for JWT method)
+
+      Public key used to validate the JWT. This key is assumed to have been pre-created using some external mechanism such as a TPM, HSM, openssl, or other method.
+
+    * --user _username_ (required)
+
+      Username of the user to add.
+
+    * --id _key_ (optional)
+
+      Optional user-specified &quot;key&quot; used for linkage with an incoming JWT via Kong&#39;s config.key\_claim\_name setting (defaults to &quot;iss&quot; field).
+
+    * --group _group_ (optional)
+
+      Group to which the user belongs, defaults to &quot;admin&quot;.
+
+    * --output-key _path_ (optional)
+
+      Output auto-generated key to a file (if --id option not supplied).
+
+  * **deluser**
+
+    Delete a proxy user. Requires additional arguments:
+
+    * --oauth (unimplemented; reserved for future use)
+
+      Delete user using OAuth authentication plugin.
+
+    * --jwt (required)
+
+      Delete user using JWT authentication plugin.
+
+    * --user _username_ (required)
+
+      Username of the user to delete.
+
+  * **jwt**
+
+    Utility function to create a JWT proxy authentication token from a supplied secret. This command does not require secret store access, but the values supplied must match those presented to the adduser command earlier. Requires additional arguments:
+
+    * --algorithm `HS256` | `ES256` (required)
+
+      Algorithm used for signing the JWT.
+
+    * --id _key_ (required)
+
+      The &quot;key&quot; field from the &quot;adduser&quot; command.
+      (This will be either the --id argument passed in, or the --output-key file contents.)
+
+    * --private\_key _private\_key.pem_ (required)
```

Comment:
Is the argument to **--private_key** the key itself, or the path to the key? If it's the latter, see my previous comment about file extension. Also are there any minimum requirements for what kind of private key (algorithm, size, ...)?

Reply:
Changed to _/path/to/private.key_ and clarified that this is PEM encoded and should match the previously specified algorithm.  I don't really want to pick algorithm sizes in the documentation.  Let's just have a gentlemen's agreement that this should be RSA-3072 or larger or ECDSA-384 or larger.  I would also advocate RS384 vs RS256, but RS384 is not supported by the Kong JWT plugin at this time.